run_unittests.py

```diff
@@ -4646,6 +4646,40 @@ def test_link_environment_variable_optlink(self):
     def test_link_environment_variable_rust(self):
         self._check_ld('link', 'rust', 'link')
 
+    def test_pefile_checksum_repro(self):
+        try:
+            import pefile
+        except ImportError:
+            if is_ci():
+                raise
+            raise unittest.SkipTest('pefile module not found')
+        testdir = os.path.join(self.common_test_dir, '6 linkshared')
+        self.init(testdir)
+        self.build()
+        # Test that binaries have a non-zero checksum
+        env = get_fake_env()
+        cc = env.detect_c_compiler(MachineChoice.HOST)
+        cc_id = cc.get_id()
+        ld_id = cc.get_linker_id()
+        dll = glob(os.path.join(self.builddir, '*mycpplib.dll'))[0]
+        exe = os.path.join(self.builddir, 'cppprog.exe')
+        for f in (dll, exe):
+            pe = pefile.PE(f)
+            # Verify that a valid checksum was written
+            self.assertTrue(pe.verify_checksum(),
+                    msg='PE file: {!r}, compiler: {!r}, linker: {!r}'.format(f, cc_id, ld_id))
```

Comment:
__[Flake8]__ continuation line under-indented for visual indent ([see on Sider](https://sider.review/gh/repos/19784232/pulls/6451#issue-5054829))

<details>

| Rule |
| - |
| E128 |

</details>
<sub>posted by Sider</sub>